framework/src/main/java/org/checkerframework/common/basetype/BaseTypeVisitor.java

```diff
@@ -914,6 +972,31 @@ public Void visitVariable(VariableTree node, Void p) {
         }
     }
 
+    /**
+     * Return true if the given annotation is a type annotation: that is, its definition is
+     * meta-annotated with {@code @Target({TYPE_USE,....})}.
+     */
+    private boolean isTypeAnnotation(AnnotationTree anno) {
```

Comment:
Thanks for pointing out that method.  That method is undocumented, has a weird interface (requires a Class<?>, but just for diagnostic messages), and crashes the Checker Framework if the user ever writes an annotation that is both a type annotation and a declaration annotation.  I think it would need to be cleaned up to be usable for clients.